### ISSUE TITLE
[FLANG][OMP] flang fails to detect invalid dependence-type for DEPEND on a TASKWAIT directive

### ISSUE THREAD

**Issue by laoshd** [llvm/llvm-project#219234]
Compiler is expected to produce an error when compiling the following test. But flang does not.

`% cat tst.f90`

```fortran
PROGRAM TST
  USE ISO_FORTRAN_ENV, ONLY: ERROR_UNIT
  USE OMP_LIB
  INTEGER :: x = 3 !Depend variable

  !$OMP PARALLEL
    !$OMP SINGLE
      !Below line should throw an error
      !$OMP TASKWAIT DEPEND(MUTEXINOUTSET:x)
    !$OMP END SINGLE
  !$OMP END PARALLEL
END
```

```console
% flang tst.f90 -fopenmp -fopenmp-version=50
warning: OpenMP support for version 50 in flang is still incomplete [-Wexperimental-option]
```

**Comment by laoshd** [comment]
I'd like to work on this. Would you please assign it to me @laoshd? Thanks.
